react-toggle-switch-component/src/toggle-switch.jsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+
+export default class ToggleSwitch extends React.Component{
+    constructor(props){
+        super(props);
+        this.state = ({switchOn:false});
+        this.handleClick = this.handleClick.bind(this);
+    }
+
+    handleClick(){
+        if(this.state.switchOn){
+            this.setState({switchOn:false});
+        }
+        else {
+            this.setState({switchOn: true});
+        }
+    }
+
+    render(){
+        if(this.state.switchOn){
+            return <button onClick={this.handleClick} className={'on'}><button className={'innerOn'}></button><div className={'text'}>ON</div></button>;
+        }
+        else {
+            return <button onClick={this.handleClick}><button className={'innerOff'}></button><div className={'text'}>OFF</div></button>;
+        }
+    }

```

Comment:
This is okay and will work, but a better way to do it would be to create one button and use `this.state.switchOn` to determine what the class names should be.